ipc/provider/src/lib.rs

```diff
@@ -572,6 +598,25 @@ impl IpcProvider {
 
         conn.manager().get_chain_id().await
     }
+
+    pub async fn get_chain_head_height(&self, subnet: &SubnetID) -> anyhow::Result<ChainEpoch> {
+        let conn = match self.connection(subnet) {
```

Comment:
Isn't the change set picked up from the parent?

Reply:
this is get chain head height? Anyways, for change set, it's from the subnet actor in parent, so the address should be derived from the subnet directly.